File: medium/beer-song.js
export const recite = (initialBottlesCount, takeDownCount) => {
  let bottles = initialBottlesCount;
  let lyrics = [];
  for (let i = 0; i < takeDownCount; i++) {
    if (i >= 1) {
      lyrics.push("");
    }
    if (bottles >= 3) {
      lyrics.push(
        `${bottles} bottles of beer on the wall, ${bottles} bottles of beer.`
      );
      lyrics.push(
        `Take one down and pass it around, ${--bottles} bottles of beer on the wall.`
      );
    } else if (bottles === 2) {
      lyrics.push(
        `${bottles} bottles of beer on the wall, ${bottles} bottles of beer.`
      );
      lyrics.push(
        `Take one down and pass it around, ${--bottles} bottle of beer on the wall.`
      );
    } else if (bottles === 1) {
      lyrics.push(
        `${bottles} bottle of beer on the wall, ${bottles} bottle of beer.`
      );
      lyrics.push(
        `Take it down and pass it around, no more bottles of beer on the wall.`
      );
      --bottles;
    } else {
      lyrics.push(
        `No more bottles of beer on the wall, no more bottles of beer.`
      );
      lyrics.push(
        `Go to the store and buy some more, 99 bottles of beer on the wall.`
      );
    }
  }
  return lyrics;
};

File: python/easy/beer_song.py
def recite(start, take=1):
    verses = []
    for i in range(take):
        if start - i > 2:
            verses.append(f"{start - i} bottles of beer on the wall, {start - i} bottles of beer.")
            verses.append(f"Take one down and pass it around, {start - i - 1} bottles of beer on the wall.")
        elif start - i == 2:
            verses.append(f"{start - i} bottles of beer on the wall, {start - i} bottles of beer.")
            verses.append(f"Take one down and pass it around, {start - i - 1} bottle of beer on the wall.")
        elif start - i == 1:
            verses.append(f"{start - i} bottle of beer on the wall, {start - i} bottle of beer.")
            verses.append("Take it down and pass it around, no more bottles of beer on the wall.")
        else:
            verses.append("No more bottles of beer on the wall, no more bottles of beer.")
            verses.append("Go to the store and buy some more, 99 bottles of beer on the wall.")
        if take > 1 and i < take - 1:
            verses.append("")

    return verses
